sdcm/cluster.py

```diff
@@ -4447,35 +4447,26 @@ def wait_for_schema_agreement(self):
         self.log.debug('Schema agreement is reached')
         return True
 
-    def check_nodes_up_and_normal(self, nodes=None, verification_node=None):
-        """Checks via nodetool that node joined the cluster and reached 'UN' state"""
+    def check_nodes_up_and_normal(self, nodes: Optional[list[BaseNode]] = None, verification_node: Optional[BaseNode] = None):
+        """Checks via nodetool executed on verification node that nodes joined the cluster and reached 'UN' state"""
         if not nodes:
             nodes = self.nodes
         status = self.get_nodetool_status(verification_node=verification_node)
-        up_statuses = []
+        ip_statuses = {key: value for dc_value in status.values() for key, value in dc_value.items()}
```

Comment:
Can we make Ip statuses in separate function to avoid code duplication?

Reply:
done